pkg/client/unversioned/helper.go

```diff
@@ -428,7 +440,13 @@ func UnversionedRESTClientFor(config *Config) (*RESTClient, error) {
 		return nil, err
 	}
 
-	client := NewRESTClient(baseURL, unversioned.SchemeGroupVersion, config.Codec, config.QPS, config.Burst)
+	versionConfig := config.ContentConfig
+	if versionConfig.GroupVersion == nil {
+		v := unversioned.SchemeGroupVersion
+		versionConfig.GroupVersion = &v
```

Comment:
We're copying unversioned.SchemeGroupVersion so we guarantee that the rest client doesn't accidentally mutate a system constant